packages/signers/src/lit-protocol/signer.ts

```diff
@@ -0,0 +1,200 @@
+import { PKPEthersWallet } from "@lit-protocol/pkp-ethers";
+import { LitAbility, LitActionResource } from "@lit-protocol/auth-helpers";
+import {
+  type AuthCallbackParams,
+  type AuthSig,
+  type SessionSigsMap,
+} from "@lit-protocol/types";
+import { LitNodeClient } from "@lit-protocol/lit-node-client";
+import { generateSessionKeyPair } from "@lit-protocol/crypto";
+
+import {
+  type SignTypedDataParams,
+  type SmartAccountAuthenticator,
+} from "@alchemy/aa-core";
+import type { TypedDataDomain } from "viem";
+
+import {
+  type LitAuthMethod,
+  type LitSessionSigsMap,
+  type LitAccountAuthenticatorParams,
+  type LITAuthenticateProps,
+} from "./types.js";
+
+const SIGNER_TYPE: string = "lit";
+
+/**
+ * Implementation of `SmartAccountAuthenticator` for lit protocol
+ * This class requies:
+ * `@lit-protocol/lit-node-client`
+ * `@lit-protocol/pkp-ethers`
+ */
+export class LitSigner<C extends LitAuthMethod | LitSessionSigsMap>
+  implements
+    SmartAccountAuthenticator<
+      LITAuthenticateProps<C>,
+      LitSessionSigsMap,
+      PKPEthersWallet | undefined
+    >
+{
+  inner: PKPEthersWallet | undefined;
+  private _signer: PKPEthersWallet | undefined;
+  private _client: LitNodeClient;
+  private _pkpPublicKey: string;
+  private _rpcUrl: string;
+  private _authContext: C | undefined;
+  private _session: SessionSigsMap | undefined;
+
+  constructor(params: LitAccountAuthenticatorParams) {
+    this._pkpPublicKey = params.pkpPublicKey;
+    this._client = new LitNodeClient({
+      litNetwork: params.network ?? "cayenne",
+      debug: params.debug ?? false,
+    });
+    this._rpcUrl = params.rpcUrl;
+  }
+  signerType: string = SIGNER_TYPE;
+
+  /**
+   * Authenticates the supplied authemtnication material if type `Auth Method`
+   * if type `SessionSigsMap` this implementation will respect the existing auth and use the session material
+   * @param props {LITAuthenticateProps} Authentication params, only `context` is required
+   * @returns {LitSessionSigsMap} Authenticated session material
+   */
+  authenticate = async (
```

Comment:
the testing is largely mocked out as the other tests seem to avoid network calls. did run tests with network I/O before replacing with mocks to confirm functionality.

Reply:
> as the other tests seem to avoid network calls
largely due to potential flakiness / need for an account. if this is something that can be run consistently without flakiness / rate-limiting then it'd be awesome to have them do network I/O